osquery/registry/registry_interface.cpp

```diff
@@ -268,29 +270,24 @@ Status RegistryInterface::addExternal(const RouteUUID& uuid,
 
 /// Remove all the routes for a given uuid.
 void RegistryInterface::removeExternal(const RouteUUID& uuid) {
+  WriteLock lock(mutex_);
```

Comment:
Let's take a look at `registry_factory.cpp` where this is called and see if there are other locks that need to be synchronized.

Reply:
- Look at calls into `::removeExternal`
- Look for the implementation of `::removeExternalPlugin`